capabilities/writetarget/beholder/monitor/beholder.go

```diff
@@ -0,0 +1,97 @@
+package monitor
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+const (
+	// Helper keys to avoid duplicating attributes
+	CtxKeySkipAppendAttrs = "skip_append_attrs"
+)
+
+// BeholderClient is a Beholder client extension with a custom ProtoEmitter
+type BeholderClient struct {
+	*beholder.Client
+	ProtoEmitter ProtoEmitter
+}
+
+// ProtoEmitter is an interface for emitting protobuf messages
+// TODO: this should be moved to chainlink-common
+type ProtoEmitter interface {
```

Comment:
This is something that is not directly tied to WT capability, and is a separate framework to emit Protobuf messages via Beholder. It should not be grouped under `/writetarget/...`

It potentially belongs in chainlink-common.

Reply:
added TODO here